src/backend/core/api/serializers.py

```diff
@@ -226,6 +229,23 @@ def validate_file(self, file):
                 f"File type '{mime_type:s}' is not allowed. Allowed types are: {mime_types:s}"
             )
 
+        # Compress the image if it’s a valid image type
+        if mime_type and mime_type.startswith("image/"):
+            original_extension = os.path.splitext(file.name)[1]
+            image = Image.open(file)
+            image_io = BytesIO()
+            image.save(image_io, format=image.format, quality=85, optimize=True)
```

Comment:
This will only optimize few image formats. Should we check the original image size then use a `.resize(` to reduce the size? 

Edit: ok, this is mentioned in the following comment ^^

Reply:
@qbey I'm not sure what's the preferred way to go as there are multiple solutions to choose from, we could directly passthrough svg images, or we could convert them to png using e.g. cairosvg

```
        # Handle SVG files using CairoSVG
        if mime_type == "image/svg+xml":
            svg_io = BytesIO()
            file.seek(0)
            cairosvg.svg2png(file_obj=file, write_to=svg_io)
            svg_io.seek(0)
            file = InMemoryUploadedFile(
                svg_io, None, f"{os.path.splitext(file.name)[0]}.png", "image/png", svg_io.getbuffer().nbytes, None
            )
```

We would then also apply the max image size and compression onto the final png created from the source svg. But I'm not sure if this is really the best way to go, as svg files have features png files don't have ... And I'm not sure if you wanted to support things like animated svg for example. As in that case, I would simply suggest to passthrough the svg file directly, just limiting the user by the file size he wants to upload (currently 10 MB).

Reply:
Hi @lindenb1 ,

As I see it, I would as you said, for SVG:
> I would simply suggest to passthrough the svg file directly, just limiting the user by the file size he wants to upload (currently 10 MB).

I don't really like the idea of converting SVG to PNG. But, this actually raises the question about security here, because SVG should be sanitized also... So maybe we should accept SVG as it is, block if above 10MB, and simply add an issue to address the security problem.

Anyway, the SVG format should be addressed separately IMHO. 

What do you think @sampaccoud ?

Reply:
@qbey I just updated my last commit to include the passthrough with svg files.

Reply:
As commented on the other discussion, I would try using a library like [easythumbnails](https://pypi.org/project/easy-thumbnails/) because resizing images probably has even more hidden complexities than what we raised here?

Reply:
so let's forget this comment on mine :ok_hand: